src/modules/fundingManager/oracle/FM_PC_ExternalPrice_Redeeming_v1.sol

```diff
@@ -0,0 +1,672 @@
+import {IRedeemingBondingCurveBase_v1} from
+    "@fm/bondingCurve/interfaces/IRedeemingBondingCurveBase_v1.sol";
+import {Module_v1} from "src/modules/base/Module_v1.sol";
+import {FM_BC_Tools} from "@fm/bondingCurve/FM_BC_Tools.sol";
+import {IERC20PaymentClientBase_v1} from
+    "@lm/interfaces/IERC20PaymentClientBase_v1.sol";
+
+// External
+import {IERC20} from "@oz/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@oz/token/ERC20/utils/SafeERC20.sol";
+import {IERC20Metadata} from "@oz/token/ERC20/extensions/IERC20Metadata.sol";
+import {IERC20Issuance_v1} from "@ex/token/ERC20Issuance_v1.sol";
+import {ERC165Upgradeable} from
+    "@oz-up/utils/introspection/ERC165Upgradeable.sol";
+
+/**
+ * @title   External Price Oracle Funding Manager with Payment Client
+ *
+ * @notice  A funding manager implementation that uses external oracle price feeds
+ *          for token operations. It integrates payment client functionality and
+ *          supports token redemption through a bonding curve mechanism.
+ *
+ * @dev     This contract inherits from:
+ *              - IFM_PC_ExternalPrice_Redeeming_v1
+ *              - ERC20PaymentClientBase_v1
+ *              - RedeemingBondingCurveBase_v1
+ *          Key features:
+ *              - External price integration
+ *              - Payment client functionality
+ *          Price feeds are managed through an external oracle contract that
+ *          implements IOraclePrice_v1.
+ *
+ * @custom:security-contact security@inverter.network
+ *                          In case of any concerns or findings, please refer to
+ *                          our Security Policy at security.inverter.network or
+ *                          email us directly!
+ *
+ * @custom:version   1.0.0
+ *
+ * @custom:standard-version  1.0.0
+ *
+ * @author  Zealynx Security
+ */
+contract FM_PC_ExternalPrice_Redeeming_v1 is
+    IFM_PC_ExternalPrice_Redeeming_v1,
+    ERC20PaymentClientBase_v1,
+    RedeemingBondingCurveBase_v1
+{
+    /// @inheritdoc ERC165Upgradeable
+    /// @param interfaceId_ The interface identifier to check support for
+    /// @return True if the interface is supported
+    function supportsInterface(bytes4 interfaceId_)
+        public
+        view
+        override(ERC20PaymentClientBase_v1, RedeemingBondingCurveBase_v1)
+        returns (bool)
+    {
+        return interfaceId_
+            == type(IFM_PC_ExternalPrice_Redeeming_v1).interfaceId
+            || super.supportsInterface(interfaceId_);
+    }
+
+    using SafeERC20 for IERC20;
+
+    //--------------------------------------------------------------------------
+    // Constants
+
+    /// @dev Value is used to convert deposit amount to 18 decimals
+    uint8 private constant EIGHTEEN_DECIMALS = 18;
+
+    /// @notice Role for whitelisted addresses
+    bytes32 public constant WHITELIST_ROLE = "WHITELIST_ROLE";
+
+    /// @notice Role for payment queue
+    bytes32 public constant PAYMENT_QUEUE_ROLE = "PAYMENT_QUEUE_ROLE";
```

Comment:
I know that I've typed the name of this role while we were in the call :D but I think it's not descriptive enough. Reason being that payment could also be considered in the context of buying. Could you please rename it to something more descriptive ser.

Maybe `"REDEMPTION_QUEUE_EXECUTOR_ROLE"`  (although this might be a bit long). Or maybe `"QUEUE_MANAGER_ROLE"` 🤔

I leave it up to you

Reply:
I chose `QUEUE_MANAGER_ROLE`